app/src/main/java/space/taran/arknavigator/mvp/presenter/GalleryPresenter.kt

```diff
@@ -219,6 +236,21 @@ class GalleryPresenter(
         viewState.showEditTagsDialog(currentResource.id)
     }
 
+    fun onIncreaseScore() = changeScore(1)
+
+    fun onDecreaseScore() = changeScore(-1)
+
+    private fun changeScore(inc: Score) = presenterScope.launch {
+        val score = scoreStorage.getScore(currentResource.id) + inc
+        scoreStorage.setScore(currentResource.id, score)
+        withContext(Dispatchers.IO) {
+            scoreStorage.persist()
+        }
+        withContext(Dispatchers.Main) {
+            viewState.displayScore(score)
+        }
```

Comment:
Here you also need to update scores on resources screen, for this you can use `viewState.notifyResourcesChanged()`

Reply:
I have implemented it